ui/src/state/chats.rs

```diff
@@ -54,3 +54,11 @@ pub enum Direction {
     Incoming,
     Outgoing,
 }
+
+impl Chats {
+    pub fn join(&mut self, other: &mut HashMap<Uuid, Chat>) {
```

Comment:
If youre planning on reusing `other`, you could just have this as a `mut ` instead of `&mut`, though i guess we could technically extend `self.all` with `other` too with `other` being moved to this function